client/src/rr/dnssec/verifier.rs

```diff
@@ -41,11 +45,15 @@ pub trait Verifier {
     ///
     /// `true` if the message could be validated against the signature, `false` otherwise
     fn verify_message(&self, message: &Message, signature: &[u8], sig0: &SIG) -> DnsSecResult<()> {
-        hash::hash_message(message, sig0).and_then(|hash| self.verify(&hash, signature))
+        tbs::message_tbs(message, sig0).and_then(|tbs| self.verify(tbs.as_ref(), signature))
     }
 
     /// Verifies an RRSig with the associated key, e.g. DNSKEY
     ///
+    /// XXX: This does the signature verification and the parsing in the wrong order; the signature
```

Comment:
Are you pointing out a bug here?

Also, do you use `XXX` for `TODO`?, Can we instead use `TODO` or `FIXME` for clarity? I use `FIXME` for things that generally should be fixed prior to merge, `TODO` for things that are ok but should be revisited in the future (performance, etc.)